Trunk

SylviaVargasCTL opened the pull request.

Commit:
KAFKA-1054; Eliminate Scala Compilation Warnings

Changes:
- Suppressed compiler warnings about type erasure in matching via unboxing
by Jon Riehl.
- Suppressed warning caused by slight difference in input function type
by John Riehl.
- Fix compiler warnings: ServerShutdownTest, DelayedJoinGroup function
signature by Blake Smith.
- Fix Scala 2.11 warnings. `Pair` has been deprecated, `try` without
`catch` and `finally` is useless and initialisation order fix by Ismael
Juma.

joestein committed:
KAFKA-1812 Allow IpV6 in configuration with parseCsvMap patch by Jeff Holoman reviewed by Gwen Shapira and Joe Stein

Commit:
kafka-1815; ServerShutdownTest fails in trunk; patched by Chris Cope; reviewed by Jun Rao

fanatoly committed:
kafka-1813; Build fails for scala 2.9.2; patched by Anatoly Fayngelerin; reviewed by Jun Rao

Commit:
KAFKA-742: Existing directories under the Kafka data directory without any data cause process to not start; reviewed by Neha Narkhede

junrao committed:
kafka-1797; add the serializer/deserializer api to the new java client; patched by Jun Rao; reviewed by Neha Narkhede

gwenshap committed:
KAFKA-1824 - fix ConsoleProducer so parse.key and key.separator will work again; reviewed by Neha Narkhede

becketqin committed:
KAFKA-1650; (Follow-up patch) to support no data loss in mirror maker; reviewed by Joel Koshy

ewencp committed:
kafka-1642; (followup patch) [Java New Producer Kafka Trunk] CPU Usage Spike to 100% when network connection is lost; patched by Ewen Cheslack-Postava; patched by Ewen Cheslack-Postava; reviewed by Jun Rao

junrao committed:
kafka-1797; (follow-up patch) add the serializer/deserializer api to the new java client; patched by Jun Rao; reviewed by Jay Kreps

junrao committed:
kafka-1797; (delta follow-up patch) add the serializer/deserializer api to the new java client; patched by Jun Rao; reviewed by Neha Narkhede

joestein committed:
KAFKA-1512 Fixes for limit the maximum number of connections per ip address patch by Jeff Holoman reviewed by Jay Krepps and Gwen Shapira

junrao committed:
kafka-1797; (addressing Manikumar Reddy's comment) add the serializer/deserializer api to the new java client; patched by Jun Rao; reviewed by Manikumar Reddy and Neha Narkhede

junrao committed:
kafka-1851; OffsetFetchRequest returns extra partitions when input only contains unknown partitions; patched by Jun Rao; reviewed by Neha Narkhede

jkreps committed:
KAFKA-1760: New consumer.

Commit:
KAFKA-1070 Auto assign broker id; reviewed by Neha Narkhede

jaiforums committed:
KAFKA-1836 metadata.fetch.timeout.ms set to zero blocks forever; reviewed by Neha Narkhede and Ewen Cheslack-Postava

junrao committed:
kafka-1797; (missed parametric in a few files) add the serializer/deserializer api to the new java client; patched by Jun Rao

gwenshap committed:
KAFKA-1819 Cleaner gets confused about deleted and re-created topics; reviewed by Neha Narkhede

omkreddy committed:
KAFKA-1723; make the metrics name in new producer more standard; patched by Manikumar Reddy; reviewed by Jay Kreps and Jun Rao

jaiforums committed:
KAFKA-1854 Allow JIRA username and password to be prompted in the absence of a jira.ini file, during patch submission; reviewed by Neha Narkhede

junrao committed:
trivial change to add byte serializer to ProducerPerformance; patched by Jun Rao

omkreddy committed:
KAFKA-1723 (delta patch to fix javadoc); make the metrics name in new producer more standard; patched by Manikumar Reddy; reviewed by Jun Rao

omkreddy committed:
KAFKA-1499; Broker-side compression configuration; reviewed by Joel Koshy

jjkoshy committed:
KAFKA-1499; Trivial follow-up (fix comments and whitespace)

junrao committed:
kafka-1868; ConsoleConsumer shouldn't override dual.commit.enabled to false if not explicitly set; patched by Jun Rao; reviewed by Jeol Koshy

junrao committed:
kafka-1870; Cannot commit with simpleConsumer on Zookeeper only with Java API; patched by Jun Rao; reviewed by Jeol Koshy and Sriharsha Chintalapani

junrao committed:
kafka-1864; Revisit defaults for the internal offsets topic; patched by Jun Rao; reviewed by Jeol Koshy, Neha Narkhede, and Gwen Shapira

junrao committed:
trivial change to remove -UseCompressedOops option from script

omkreddy committed:
KAFKA-1723; num.partitions documented default is 1 while actual default is 2; patched by Manikumar Reddy; reviewed by Jun Rao

jaiforums committed:
KAFKA-1878; ProducerFailureHandlingTest.testCannotSendToInternalTopic fails with TimeoutException while trying to fetch metadata for topic; patched by jaikiran pai; reviewed by Jun Rao

junrao committed:
KAFKA-1876; pom file for scala 2.11 should reference a specific version; patched by Jun Rao; reviewed by Joe Stein and Sriharsha Chintalapani

jjkoshy committed:
KAFKA-1823; Fix transient failure in PartitionAssignorTest; reviewed by Guozhang Wang and Neha Narkhede

auradkar committed:
KAFKA-1848; check consumer shutting down flag inside retry loop; reviewed by Guozhang Wang

joestein committed:
KAFKA-1891 MirrorMaker hides consumer exception - making troubleshooting challenging patch by Gwen Shapira reviewed by Joe Stein

becketqin committed:
KAFKA-1890 Fix bug preventing Mirror Maker from successful rebalance; reviewed by Gwen Shapira and Neha Narkhede

becketqin committed:
KAFKA-1896; Record size function should check if value is null; reviewed by Guozhang Wang

omkreddy committed:
KAFKA-1109 Need to fix GC log configuration code, not able to override KAFKA_GC_LOG_OPTS; reviewed by Neha Narkhede

jaiforums committed:
KAFKA-1883 Fix NullPointerException in RequestSendThread; reviewed by Neha Narkhede

jaiforums committed:
KAFKA-1885 Upgrade junit dependency in core to 4.6 version to allow running individual test methods via gradle command line; reviewed by Neha Narkhede

olandere committed:
KAFKA-1818 KAFKA-1818 clean up code to more idiomatic scala usage; reviewed by Neha Narkhede and Gwen Shapira

junrao committed:
KAFKA-1902; fix MetricName so that Yammer reporter can work correctly; patched by Jun Rao; reviewed by Manikumar Reddy, Manikumar Reddy and Joel Koshy

omkreddy committed:
KAFKA-1861; Publishing kafka-client:test in order to utilize the helper utils in TestUtils; patched by Manikumar Reddy; reviewed by Jun Rao

jkreps committed:
KAFKA-1760 Follow-up: fix compilation issue with Scala 2.11

jjkoshy committed:
KAFKA-1729; Add constructor to javaapi to allow constructing explicitly versioned offset commit requests; reviewed by Jun Rao

jkreps committed:
KAFKA-1915: Add checkstyle for java code.

jkreps committed:
KAFKA-1919: Always update the metadata, when a metadata response is received to ensure we back off.

jkreps committed:
KAFKA-1865 Add a flush() method to the producer.

guozhangwang committed:
KAFKA-1925; Fix coordinator broker id stuck with INT_MIN; reviewed by Jay Kreps

guozhangwang committed:
KAFKA-1333; Add the consumer coordinator to server; reviewed by Onur Karaman and Jay Kreps

guozhangwang committed:
KAFKA-1333 follow-up; Add missing files for the coordinator folder

onurkaraman committed:
KAFKA-1476 Added a ConsumerCommand tool that will replace other consumer related tools in the future; reviewed by Neha Narkhede

guozhangwang committed:
KAFKA-1948; Fix ConsumerTest.testPartitionReassignmentCallback handling issue; reviewed by Gwen Shapira

gwenshap committed:
KAFKA-1697; Remove support for producer ack > 1 on the broker; reviewed by Joel Koshy

Commit:
KAFKA-1805; ProducerRecord should implement equals and hashCode; reviewed by Guozhang Wang

auradkar committed:
KAFKA-1943; MessageSizeTooLarge and MessageSetSizeTooLarge should not be counted toward broker-side producer failure rate

auradkar committed:
KAFKA-1914; Include total produce/fetch stats in broker topic metrics.

Commit:
KAFKA-1959; Rename group to groupId in TestOffsetManager due to collision with Thread.group in IBM's JDK; reviewed by Joel Koshy and Gwen Shapira

Commit:
KAFKA-1960; .gitignore does not exclude test generated files and folders; reviewed by Joel Koshy and Gwen Shapira

junrao committed:
kafka-1952; High CPU Usage in 0.8.2 release; patched by Jun Rao; reviewed by Guozhang Wang, Ewen Cheslack-Postava and Neha Narkhede

jjkoshy committed:
KAFKA-1914; follow-up to address unit test failure

auradkar committed:
KAFKA-1914; follow-up to fix SimpleFetchTest; reviewed by Joel Koshy

jjkoshy committed:
KAFKA-1729; (followup patch)Add constructor to javaapi to allow
constructing explicitly versioned offset fetch requests; reviewed by Jun Rao

zhouyaguo committed:
Minor typo in KafkaConfig; reviewed by Gwen Shapira

Commit:
KAFKA-1867 liveBroker list not updated on a cluster with no topics; reviewed by Neha Narkhede

ymatsuda committed:
Leaner DelayedItem; reviewed by Neha Narkhede and Joel Koshy

Commit:
KAFKA-1947 can't explicitly set replica-assignment when add partitions; reviewed by Neha Narkhede

jkreps committed:
KAFKA-1757 Unmap file prior to delete on windows.

junrao committed:
kafka-1971; starting a broker with a conflicting id will delete the previous broker registration; patched by Jun Rao; reviewed by Neha Narkhede

onurkaraman committed:
Added script for bin/kafka-consumer-groups.sh that was mistakenly dropped from the original patch for KAFKA-1476

junrao committed:
kafka-1984; java producer may miss an available partition; patched by Jun Rao; reviewed by Ewen Cheslack-Postava, Jay Kreps, and Guozhang Wang

jjkoshy committed:
KAFKA-1953; KAFKA-1962; Disambiguate purgatory metrics; restore delayed request metrics; reviewed by Guozhang Wang

ewencp committed:
kafka-1881; transient unit test failure in testDeleteTopicWithCleaner due to OOME; patched by Ewen Cheslack-Postava; reviewed by Jun Rao

junrao committed:
kafka-1400; transient unit test failure in SocketServerTest; patched by Jun Rao; reviewed by Ewen Cheslack-Postava and Jiangjie Qin

nehanarkhede committed:
Deleting the ConsumerTest until the issue with the hanging test is resolved; discussed on the mailing list and got several +1s

Commit:
KAFKA-1664 Kafka does not properly parse multiple ZK nodes with non-root chroot; reviewed by Neha Narkhede and Jun Rao

Commit:
KAFKA-1866 LogStartOffset gauge throws exceptions after log.delete(); reviewed by Neha Narkhede

gwenshap committed:
KAFKA-1824 ConsoleProducer - properties key.separator and parse.key no longer work; reviewed by Neha Narkhede

jjkoshy committed:
KAFKA-1755; Reject compressed and unkeyed messages sent to compacted topics; reviewed by Mayuresh Gharat, Neha Narkhede and Guozhang Wang

Commit:
KAFKA-1852; Reject offset commits to unknown topics; reviewed by Joel Koshy

jjkoshy committed:
KAFKA-1499; trivial follow-up (remove unnecessary parentheses)

auradkar committed:
KAFKA-1986; Request failure rate should not include invalid message size and offset out of range; reviewed by Joel Koshy

jjkoshy committed:
KAFKA-2001; Trivial commit to prevent OffsetCommitTest from hanging

jjkoshy committed:
KAFKA-2001; Trivial commit to fix OffsetCommitTest

gwenshap committed:
kafka-1999; Fix failing unit-test: kafka.api.ProducerFailureHandlingTest > testNotEnoughReplicasAfterBrokerShutdown; patched by Gwen Shapira; reviewed by Jun Rao

jjkoshy committed:
KAFKA-1755; trivial follow-up to fix comment in CleanerTest

Commit:
KAFKA-1988; Fix org.apache.kafka.common.utils.Utils.abs and add Partitioner.toPositive; reviewed by Jun Rao and Guozhang Wang

joestein committed:
KAFKA-1845 KafkaConfig should use ConfigDef patch by Andrii Biletskyi reviewed by Gwen Shapira

becketqin committed:
KAFKA-2009 Fix two minor bugs in mirror maker.

guozhangwang committed:
KAFKA-1910; Refactor new consumer and fixed a bunch of corner cases / unit tests; reviewed by Onur Karaman and Jay Kreps

guozhangwang committed:
KAFKA-1910; follow-up on fixing buffer.flip on produce requests

guozhangwang committed:
KAFKA-1910; missed follow-up changes

guozhangwang committed:
KAFKA-1910 Follow-up again; fix ListOffsetResponse handling for the expected error codes

Commit:
kafka-1461; Replica fetcher thread does not implement any back-off behavior; patched by Sriharsha Chintalapani; reviewed by Jun Rao

becketqin committed:
KAFKA-1997; Refactor MirrorMaker based on KIP-3; reviewed by Joel Koshy and Guozhang Wang

guozhangwang committed:
KAFKA-1863; Add docs for possible thrown exception in Callback; reviewed by Jiangjie Qin

becketqin committed:
KAFKA-1997; Follow-up to add the shutdown hook before starting the consumers; reviewed by Guozhang Wang

becketqin committed:
KAFKA-1997; Hopefully last follow-up fix to get messageHandlerArgs right

guozhangwang committed:
KAFKA-1910; Fix two bugs on MemoryRecords and KafkaConsumer; reviewed by Onur Karaman

guozhangwang committed:
trivial fix on coding style

becketqin committed:
KAFKA-2042; Update topic list of the metadata regardless of cluster information; reviewed by Guozhang Wang

xiaotao183 committed:
KAFKA-2048; Change lock synchronized to inLock() for partitionMapCond; reviewed by Guozhang Wang

joestein committed:
KAFKA-1856 Add PreCommit Patch Testing patch by Ashish K Singh reviewed by Gwen Shapira

ymatsuda committed:
kafka-527; Compression support does numerous byte copies; patched by Yasuhiro Matsuda; reviewed by Guozhang Wang and Jun Rao

becketqin committed:
KAFKA-2047; Move the stream creation into concurrent mirror maker threads; reviewed by Guozhang Wang

guozhangwang committed:
KAFKA-527; Use in-place decompression enabled inner iterator to replace old decompress function; reviewed by Joel Koshy and Jun Rao

guozhangwang committed:
KAFKA-1634; Bump up Offset Commit Request to v2 to add global retention and remove per-partition commit timestamp; reviewed by Joel Koshy and Jun Rao

gwenshap committed:
kafka-2044; Support requests and responses from o.a.k.common in KafkaApis; patched by Gwen Shapira; reviewed by Jun Rao

ijuma committed:
kafka-2039; Update Scala to 2.10.5 and 2.11.6; patched by Ismael Juma; reviewed by Jun Rao

ymatsuda committed:
kafka-2013; benchmark test for the purgatory; patched by Yasuhiro Matsuda; reviewed by Jun Rao

Commit:
kafka-2016; RollingBounceTest takes long; patched by Ted Malaska; reviewed by Jun Rao

Commit:
KAFKA-1961 Prevent deletion of _consumer_offsets topic; reviewed by Neha Narkhede, Gwen Shapira and Jun Rao

auradkar committed:
KAFKA-1546; Automate replica lag tuning; reviewed by Joel Koshy, Neha
Narkhede, Jun Rao and Guozhang Wang

ewencp committed:
KAFKA-1501 Let the OS choose the port in unit tests to avoid collisions. Patch by Ewen CP, reviewed by Guozhang and me.

Commit:
KAFKA-2050 Avoid calling .size() on linked list.

zhouyaguo committed:
KAFKA-2002: Mx4JLoader doesn't disable when kafka_mx4jenable=false.

rajinisivaram committed:
KAFKA-2024 Log compaction can generate unindexable segments.

zhouyaguo committed:
KAFKA-1996 Fix scaladoc error.

pmackles committed:
KAFKA-1005 Shutdown consumer at the end of consumer performance test.

gwenshap committed:
kafka-1809; Refactor brokers to allow listening on multiple ports and IPs; patched by Gwen Shapira; reviewed by Joel Koshy and Jun Rao

Commit:
kafka-1926; Replace kafka.utils.Utils with o.a.k.common.utils.Utils; patched by Tong Li; reviewed by Jun Rao

granthenke committed:
kafka-2043; CompressionType is passed in each RecordAccumulator append; patched by Grant Henke; reviewed by Jun Rao

gwenshap committed:
kafka-2099; BrokerEndPoint file, methods and object names should match; patched by Gwen Shapira; reviewed by Sriharsha Chintalapani and Jun Rao

pyr committed:
kafka-2033; Small typo in documentation; patched by Pierre-Yves Ritschard; reviewed by Jun Rao

gwenshap committed:
kafka-1992; checkEnoughReplicasReachOffset doesn't need to get requiredAcks; patched by Gwen Shapira; reviewed by Jeff Holoman, Jiangjie Qin and Jun Rao

Commit:
KAFKA-1461; Implement per-partition back-off for replica fetcher; reviewed by Jun Rao and Guozhang Wang

guozhangwang committed:
KAFKA-1910 Follow-up; Revert the no-offset-committed error code; reviewed by Joel Koshy

granthenke committed:
kafka-1973; Remove the accidentally created LogCleanerManager.scala.orig; patched by Grant Henke; reviewed by Jun Rao

danielcompton committed:
kafka-1517; Messages is a required argument to Producer Performance Test; patched by Daniel Compton; reviewed by Jun Rao

gwenshap committed:
kafka-2104; testDuplicateListeners() has a typo; patched by Gwen Shapira; reviewed by Sriharsha Chintalapani, Onur Karaman and Jun Rao

Commit:
kafka-2109; Support retries in KafkaLog4jAppender; patched by Dave Beech; reviewed by Jun Rao

ymatsuda committed:
kafka-1989; New purgatory design; patched by Yasuhiro Matsuda; reviewed by Guozhang Wang and Jun Rao

Commit:
kafka-2096; Enable keepalive socket option for broker to prevent socket leak; patched by Allen Wang; reviewed by Jun Rao

Tim-Brooks committed:
kafka-2090; Remove duplicate check to metadataFetchInProgress; patched by Tim Brook; reviewed by Ewen Cheslack-PostavaJun Rao

ymatsuda committed:
kafka-2112; make overflowWheel volatile; patched by Yasuhiro Matsuda; reviewed by Jun Rao

ewencp committed:
trivial fix for compilation error in scala 2.9.1

gwenshap committed:
kafka-2115; Error updating metrics in RequestChannel; patched by Gwen Shapira; reviewed by Jun Rao

guozhangwang committed:
KAFKA-2117; Use the correct metadata field for reading offset struct; reviewed by Jun Rao and Jiangjie Qin

futtre committed:
KAFKA-1416; Unify sendMessages in TestUtils; reviewed by Guozhang Wang

lvfangmin committed:
KAFKA-2056; Fix transient testRangePartitionAssignor failure; reviewed by Guozhang Wang

rajinisivaram committed:
kafka-2113; TestPurgatoryPerformance does not compile using IBM JDK; patched by Rajini Sivaram; reviewed by Yasuhiro Matsuda and Jun Rao

ewencp committed:
kafka-2119; ConsumerRecord key() and value() methods should not have throws Exception; patched by Ewen Cheslack-Postava; reviewed by Jun Rao

Zhiqiang-He committed:
kafka-2088; kafka-console-consumer.sh should not create zookeeper path when no brokers found and chroot was set in zookeeper.connect; patched by Zhiqiang He; reviewed by Gwen Shapira and Jun Rao

Commit:
kafka-1994; Evaluate performance effect of chroot check on Topic creation; patched by Ashish Singh; reviewed by Gwen Shapira and Jun Rao

Commit:
kafka-1982; change kafka.examples.Producer to use the new java producer; patched by Ashish Singh; reviewed by Gwen Shapira, Mayuresh Gharat and Jun Rao

Commit:
kafka-1982; (add missing files) change kafka.examples.Producer to use the new java producer; patched by Ashish Singh; reviewed by Gwen Shapira, Mayuresh Gharat and Jun Rao

omkreddy committed:
kafka-2131; Update new producer javadocs with correct documentation links; patched by Manikumar Reddy; reviewed by Jun Rao

jholoman committed:
kafka-1990; Add unlimited time-based log retention; patched by Jeff Holoman; reviewed by Jun Rao

auradkar committed:
TopicConfigManager javadoc references incorrect paths; reviewed by Neha Narkhede

rajinisivaram committed:
DelayedOperationTest.testRequestExpiry transient failure; reviewed by Neha Narkhede

stevenzwu committed:
KAFKA-2121; Close internnal modules upon client shutdown; reviewed by Ewen Cheslack-Postava and Guozhang Wang

becketqin committed:
KAFKA-2138; Fix producer to honor retry backoff; reviewed by Joel Koshy and Guozhang Wang

nehanarkhede committed:
KAFKA-2034 sourceCompatibility not set in Kafka build.gradle; reviewed by Neha Narkhede and Ewen Cheslack-Postava

ijuma committed:
KAFKA-2140 Improve code readability; reviewed by Neha Narkhede

Commit:
KAFKA-2122 Remove controller.message.queue.size Config; reviewed by Neha Narkhede, Jun Rao, Onur

ijuma committed:
KAFKA-2140 follow up, checking in newly renamed file ConsumerRebalanceFailedException

sasakitoa committed:
KAFKA-2128 kafka.Kafka should return non-zero exit code when caught exception; reviewed by Neha

gwenshap committed:
KAFKA-2114 Unable to change min.insync.replicas default; reviewed by Neha

rajinisivaram committed:
kafka-2118; Cleaner cannot clean after shutdown during replaceSegments; patched by Rajini Sivaram; reviewed by Jun Rao

rekhajoshm committed:
KAFKA-1621 : Standardize --messages option. Closes #58

guozhangwang committed:
KAFKA-2121 Follow-up: minor bug fix as pointed out by Sean Lydon

Commit:
kafka-1940; Initial checkout and build failing; patched by Martin Lemanski; reviewed by Jun Rao

guozhangwang committed:
KAFKA-2068 Phase 1; Merge in KAFKA-1841; reviewed by Jun Rao

auradkar committed:
KAFKA-1886 SimpleConsumer swallowing ClosedByInterruptException; reviewed by Neha Narkhede

Commit:
KAFKA-2153 kafka-patch-review tool uploads a patch even if it is empty; reviewed by Neha Narkhede, Gwen Shapira

omkreddy committed:
KAFKA-1884; Add logging upon metadata response errors; reviewed by Guozhang Wang

stevenzwu committed:
KAFKA-2121; Fix Closeable backward-compatibility; reviewed by Guozhang Wang

guozhangwang committed:
KAFKA-2121; add missing file

Commit:
KAFKA-2150; move partitionMapCond.await into partitionMapLock; reviewed by Guozhang Wang

Commit:
KAFKA-2175; Change INFO to DEBUG for socket close on no error; and partition reassignment JSON; reviewed by Gwen Shapira and Joel Koshy

jjkoshy committed:
KAFKA-2163; The offsets manager's stale-offset-cleanup and offset load should be mutually exclusive; reviewed by Jun Rao

becketqin committed:
KAFKA-1660; Add API to the producer to support close with a timeout; reviewed by Joel Koshy and Jay Kreps.

onurkaraman committed:
KAFKA-1334; Add the heartbeat logic to consumer coordinator; reviewed by Guozhang Wang

Commit:
kafka-2169; Upgrade to zkclient-0.5; patched by Parth Brahmbhatt; reviewed by Jun Rao

guozhangwang committed:
KAFKA-2160; remove watcher list if empty in purgatory; remove join-group purgatory in coordinator; reviewed by Jun Rao

onurkaraman committed:
KAFKA-2196; Remove identical topic constraint in round-robin assignor; reviewed by Guozhang Wang

jjkoshy committed:
KAFKA-1374; Log cleaner should be able to handle compressed messages; reviewed by Guozhang Wang

becketqin committed:
KAFKA-2190; Flush mirror maker before commiting offsets; abort the
mirror-maker producer with close(0) on send error; reviewed by Joel
Koshy

becketqin committed:
KAFKA-2186; Follow-up to KAFKA-1650 - add selective offset commit to
consumer connector API; reviewed by Joel Koshy

jjkoshy committed:
Revert "KAFKA-2186; Follow-up to KAFKA-1650 - add selective offset commit to"

This reverts commit 23ff851f30bb55e794aefd2fae5367845d9230ee.

vivekpm committed:
KAFKA-1737; Enforce ZKSerializer while creating ZkClient; reviewed by Guozhang Wang

ijuma committed:
kafka-2189; Snappy compression of message batches less efficient in 0.8.2.1; patched by Ismael Juma; reviewed by Jun Rao

becketqin committed:
KAFKA-2186; Follow-up to KAFKA-1650 - add selective offset commit to consumer connector API; reviewed by Joel Koshy

Commit:
KAFKA-2091; Expose a partitioner interface in the new producer
(https://cwiki.apache.org/confluence/display/KAFKA/KIP-+22+-+Expose+a+Partitioner+interface+in+the+new+producer);
reviewed by Joel Koshy and Jay Kreps

ijuma committed:
kafka-2185; Update to Gradle 2.4; patched by Ismael Juma; reviewed by Jun Rao

ewencp committed:
KAFKA-2199 Make signing artifacts optional and disabled by
 default for SNAPSHOTs and allow remote Maven repository configuration from
 the command line.

ymatsuda committed:
kafka-2226; NullPointerException in TestPurgatoryPerformance; patched by Yasuhiro Matsuda; reviewed by Onur Karaman, Guozhang Wang and Jun Rao

ewencp committed:
KAFKA-2161; Fix a few copyrights

guozhangwang committed:
KAFKA-2208; add consumer side error handling upon coordinator failure; reviewed by Onur Karaman

gwenshap committed:
kafka-1928; Move kafka.network over to using the network classes in org.apache.kafka.common.network; patched by Gwen Shapira; reviewed by Joel Koshy, Jay Kreps, Jiangjie Qin, Guozhang Wang and Jun Rao

becketqin committed:
KAFKA-2246; UnknownTopicOrPartitionException should be an instance of InvalidMetadataException; reviewed by Ewen Cheslack-Postava and Joel Koshy

guozhangwang committed:
KAFKA-2253; fix deadlock between removeWatchersLock and watcher operations list lock; reviewed by Onur Karaman and Jiangjie Qin

Commit:
kafka-2005; Generate html report for system tests; patched by Ashish Singh; reviewed by Jun Rao

Commit:
kafka-2266; Client Selector can drop idle connections without notifying NetworkClient; patched by Jason Gustafson; reviewed by Jun Rao

apakulov committed:
kafka-2232; make MockProducer generic; patched by Alexander Pakulov; reviewed by Jun Rao

resetius committed:
kafka-2164; ReplicaFetcherThread: suspicious log message on reset offset; patched by Alexey Ozeritski; reviewed by Jun Rao

Tim-Brooks committed:
kafka-2101; Metric metadata-age is reset on a failed update; patched by Tim Brooks; reviewed by Jun Rao

abiletskyi committed:
kafka-2195; Add versionId to AbstractRequest.getErrorResponse and AbstractRequest.getRequest; patched by Andrii Biletskyi; reviewed by Jun Rao

proneetv committed:
kafka-2270; incorrect package name in unit tests; patched by Proneet Verma; reviewed by Jun Rao

Commit:
kafka-2272; listeners endpoint parsing fails if the hostname has capital letter; patched by Sriharsha Chintalapani; reviewed by Jun Rao

omkreddy committed:
kafka-2264; SESSION_TIMEOUT_MS_CONFIG in ConsumerConfig should be int; patched by Manikumar Reddy; reviewed by Jun Rao

gwenshap committed:
kafka-2252; Socket connection closing is logged, but not corresponding opening of socket; patched by Gwen Shapira; reviewed by Jun Rao

omkreddy committed:
kafka-2262; LogSegmentSize validation should be consistent; patched by Manikumar Reddy; reviewed by Jun Rao

junrao committed:
trivial fix for stylecheck error on Jenkins

gwenshap committed:
kafka-2249; KafkaConfig does not preserve original Properties; patched by Gwen Shapira; reviewed by Jun Rao

omkreddy committed:
kafka-2265; creating a topic with large number of partitions takes a long time; patched by Manikumar Reddy; reviewed by Jun Rao

omkreddy committed:
kafka-2234; Partition reassignment of a nonexistent topic prevents future reassignments; patched by Manikumar Reddy; reviewed by Jun Rao

junrao committed:
trivial change to fix unit test failure introduced in kafka-2234

omkreddy committed:
kafka-1758; corrupt recovery file prevents startup; patched by Manikumar Reddy; reviewed by Neha Narkhede and Jun Rao

Commit:
kafka-1646; Improve consumer read performance for Windows; patched by Honghai Chen; reviewed by Jay Kreps and Jun Rao

omkreddy committed:
kafka-2012; Broker should automatically handle corrupt index files;  patched by Manikumar Reddy; reviewed by Jun Rao

Commit:
kafka-2290; OffsetIndex should open RandomAccessFile consistently; patched by Chris Black; reviewed by Jun Rao

isimonenko committed:
kafka-2235; LogCleaner offset map overflow; patched by Ivan Simoneko; reviewed by Jun Rao

onurkaraman committed:
KAFKA-2245; Add response tests for consumer coordinator; reviewed by Joel Koshy

auradkar committed:
KAFKA-2293; Fix incorrect format specification in Partition.scala; reviewed by Joel Koshy

Commit:
kafka-2168; New consumer poll() can block other calls like position(), commit(), and close() indefinitely; patched by Jason Gustafson; reviewed by Jay Kreps, Ewen Cheslack-Postava, Guozhang Wang and Jun Rao

jmax01 committed:
KAFKA-2294; javadoc compile error due to illegal <p/> , build failing (jdk 8); patched by Jeff Maxwell; reviewed by Jakob Homan

xiaotao183 committed:
KAFKA-2281: avoid unnecessary value copying if logAsString is false; reviewed by Guozhang Wang

hachikuji committed:
KAFKA-2168: minor follow-up patch; reviewed by Guozhang Wang

guozhangwang committed:
KAFKA-1740: merge offset manager into consumer coordinator; reviewed by Onur Karaman and Jason Gustafson

ewencp committed:
kafka-2248; Use Apache Rat to enforce copyright headers; patched by Ewen Cheslack-Postava; reviewed by Gwen Shapira, Joel Joshy and Jun Rao

Commit:
kafka-2132; Move Log4J appender to a separate module; patched by Ashish Singh; reviewed by Gwen Shapira, Aditya Auradkar and Jun Rao

lazyval committed:
KAFKA-2314: proper MirrorMaker's message handler help message; reviewed by Guozhang Wang

Commit:
kafka-1367; Broker topic metadata not kept in sync with ZooKeeper; patched by Ashish Singh; reviewed by Jun Rao

joestein committed:
KAFKA-2304 Supported enabling JMX in Kafka Vagrantfile patch by Stevo Slavic reviewed by Ewen Cheslack-Postava

lindong28 committed:
KAFKA-2306: add another metric for buffer exhausted; reviewed by Guozhang Wang

Commit:
KAFKA-2317: follow-up of KAFKA1367; reviewed by Guozhang Wang

onurkaraman committed:
KAFKA-2313: javadoc fix for KafkaConsumer deserialization; reviewed by Guozhang Wang

lindong28 committed:
KAFKA-2298; Client Selector can drop connections on InvalidReceiveException without notifying NetworkClient; reviewed by Jason Gustafson and Joel Koshy

jjkoshy committed:
Trivial commit - explicitly exclude build/rat-report.xml from rat check

gwenshap committed:
KAFKA-2308: make MemoryRecords idempotent; reviewed by Guozhang Wang

Commit:
KAFKA-2316: Drop java 1.6 support; patched by Sriharsha Chintalapani reviewed by Ismael Juma and Gwen Shapira

Commit:
KAFKA-2327; broker doesn't start if config defines advertised.host but not advertised.port

Added unit tests as well. These fail without the fix, but pass with the fix.

Author: Geoff Anderson <geoff@confluent.io>

Closes #73 from granders/KAFKA-2327 and squashes the following commits:

52a2085 [Geoff Anderson] Cleaned up unecessary toString calls
23b3340 [Geoff Anderson] Fixes KAFKA-2327

hachikuji committed:
KAFKA-1740 follow-up: add state checking in handling heartbeat request; reviewed by Onur Karaman, Ewen Cheslack-Postavam and Guozhang Wang

jjkoshy committed:
Trivial commit - explicitly exclude build/** from rat check

Tim-Brooks committed:
KAFKA-2312: use atomic long for thread id reference; reviewed by Ewen Cheslack-Postava, Jason Gustafson, Ismael Juma and Guozhang Wang

Commit:
kafka-972; MetadataRequest returns stale list of brokers; patched by Ashish Singh; reviewed by Jun Rao

omkreddy committed:
KAFKA-2198: kafka-topics.sh exits with 0 status on failures; patched by Manikumar Reddy reviewed by Gwen Shapira

hachikuji committed:
KAFKA-2123: add callback in commit api and use a delayed queue for async requests; reviewed by Ewen Cheslack-Postava and Guozhang Wang

hachikuji committed:
KAFKA-2335; fix comment about thread safety

Author: Jason Gustafson <jason@confluent.io>

Closes #78 from hachikuji/KAFKA-2335 and squashes the following commits:

c697998 [Jason Gustafson] KAFKA-2335; fix comment about consumer thread safety

hachikuji committed:
[MINOR] fix new consumer heartbeat reschedule bug

This commit fixes a minor issue introduced in the patch for KAFKA-2123. The schedule method requires the time the task should be executed, not a delay.

Author: Jason Gustafson <jason@confluent.io>

Closes #79 from hachikuji/KAFKA-2123-fix and squashes the following commits:

6eb7ec6 [Jason Gustafson] [Minor] fix new consumer heartbeat reschedule bug

Commit:
KAFKA-2032: validate consumer's partition-assignment config; reviewed by Jason Rosenberg, Sriharsha Chintalapani and Guozhang Wang

ijuma committed:
kafka-2323; Simplify ScalaTest dependency versions; patched by Ismael Juma; reviewed by Jun Rao

lindong28 committed:
kafka-2241; AbstractFetcherThread.shutdown() should not block on ReadableByteChannel.read(buffer); patched by Dong Lin; reviewed by Jun Rao

resetius committed:
kafka-2174; Wrong TopicMetadata deserialization; patched by Alexey Ozeritskiy; reviewed by Jun Rao

ijuma commented:
Can you please close this pull request? It's triggering builds in the CI server and we cannot close it ourselves without asking Apache Infra. Thanks!

Commit:
kafka-2271; transient unit test failure in KafkaConfigConfigDefTest.testFromPropsToProps; patched by Jason Gustafson; reviewed by Jun Rao

ewencp committed:
kafka-2103; kafka.producer.AsyncProducerTest failure; patched by Ewen Cheslack-Postava; reviewed by Jun Rao

ijuma committed:
KAFKA-2324; Update to Scala 2.11.7

Author: Ismael Juma <ismael@juma.me.uk>

Closes #82 from ijuma/kafka-2324 and squashes the following commits:

d71bf5c [Ismael Juma] KAFKA-2324; Update to Scala 2.11.7

Commit:
KAFKA-2345;  Attempt to delete a topic already marked for deletion throws ZkNodeExistsException; patched by Ashish Singh; reviewed by Sriharsha Chintalapani and Ismael Juma

gwenshap committed:
Adding a file missed while committing KAFKA-2345
